File: app/src/main/java/br/com/disapps/homepet/util/ConnectionUtils.java
package br.com.disapps.homepet.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by diefferson.santos on 23/08/17.
 */

public class ConnectionUtils {

    public static int TYPE_WIFI = 1;
    public static int TYPE_MOBILE = 2;
    public static int TYPE_NOT_CONNECTED = 0;

    public static int getConnectivityStatus(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (null != activeNetwork) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI)
                return TYPE_WIFI;

            if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE)
                return TYPE_MOBILE;
        }
        return TYPE_NOT_CONNECTED;
    }


    public static boolean isOnline(Context context) {
        int status = getConnectivityStatus(context);
        return (status == TYPE_WIFI || status == TYPE_MOBILE);

//        ConnectivityManager cm =
//                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
//        NetworkInfo netInfo = cm.getActiveNetworkInfo();
//        return netInfo != null && netInfo.isConnectedOrConnecting();
    }

}

File: presentation/src/main/java/br/com/disapps/homepet/util/ConnectionUtils.kt
package br.com.disapps.homepet.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

/**
 * Created by diefferson.santos on 23/08/17.
 */
object ConnectionUtils {

    internal var TYPE_WIFI = 1
    internal var TYPE_MOBILE = 2
    internal var TYPE_NOT_CONNECTED = 0

    internal fun getConnectivityStatus(context: Context): Int {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val activeNetwork = cm.activeNetworkInfo
        if (null != activeNetwork) {
            if (activeNetwork.type == ConnectivityManager.TYPE_WIFI)
                return TYPE_WIFI

            if (activeNetwork.type == ConnectivityManager.TYPE_MOBILE)
                return TYPE_MOBILE
        }
        return TYPE_NOT_CONNECTED
    }


    fun isOnline(context: Context): Boolean {
        val status = getConnectivityStatus(context)
        return status == TYPE_WIFI || status == TYPE_MOBILE
    }
}
